src/main/kotlin/org/opensearch/replication/action/resume/TransportResumeIndexReplicationAction.kt

```diff
@@ -132,27 +128,33 @@ class TransportResumeIndexReplicationAction @Inject constructor(transportService
         var isResumable = true
         val remoteClient = client.getRemoteClusterClient(params.leaderAlias)
         val shards = clusterService.state().routingTable.indicesRouting().get(params.followerIndexName)?.shards()
-        val retentionLeaseHelper = RemoteClusterRetentionLeaseHelper(clusterService.clusterName.value(), remoteClient)
-        shards?.forEach {
-            val followerShardId = it.value.shardId
-            if  (!retentionLeaseHelper.verifyRetentionLeaseExist(ShardId(params.leaderIndex, followerShardId.id), followerShardId)) {
-                isResumable = false
+        val retentionLeaseHelper = RemoteClusterRetentionLeaseHelper(clusterService.state().metadata.clusterUUID() + ":" + clusterService.clusterName.value(), remoteClient)
+        val oldRetentionLeaseHelper = RemoteClusterRetentionLeaseHelper(clusterService.clusterName.value(), remoteClient)
+
+        if (shards != null) {
+            for(shard in shards){
+                val followerShardId = shard.value.shardId
+                if  (!retentionLeaseHelper.verifyRetentionLeaseExist(ShardId(params.leaderIndex, followerShardId.id), followerShardId)) {
+                    //Remove retention lease taken during checking for existence
+                    retentionLeaseHelper.attemptRetentionLeaseRemoval(ShardId(params.leaderIndex, followerShardId.id), followerShardId)
+                    //Checking if old retention lease id is present
+                    if(!oldRetentionLeaseHelper.verifyRetentionLeaseExist(ShardId(params.leaderIndex, followerShardId.id), followerShardId)){
+                        //old retention lease id also not present
+                        oldRetentionLeaseHelper.attemptRetentionLeaseRemoval(ShardId(params.leaderIndex, followerShardId.id), followerShardId)
+                        return false
+                    }else{
+                        //Old retention lease ID present, have to take new lease ID
```

Comment:
If the cluster had old retention lease ID and was paused for more than 12 hours, retention lease would've expired in this case. Resume should simply fail with retention lease not found. With new changes, we'll just end up adding the new retention lease instead. 


I think here we should simply verify that either old or new retention lease is present on the cluster. 
If both are not present, we should throw  `RetentionLeaseNotFoundException`. 
If old one is present, we can add the new retention lease.

Reply:
ack